web/documentserver-example/ruby/app/views/home/editor.html.erb

```diff
@@ -184,10 +185,38 @@
           innerAlert(response.error)
           return
         }
-        document.location.reload();
+        fileData = <%= raw @file.to_json %>;
+        const req = new XMLHttpRequest()
+        req.open("POST", '/historyobj')
+        req.send(JSON.stringify(fileData))
+        req.onload = function () {
+            versionHistory = JSON.parse(req.response)
+            docEditor.refreshHistory(versionHistory.hist)
+        }
       }
     }
 
+    function onRequestHistory() {
+        fileData = <%= raw @file.to_json %>;
+        const req = new XMLHttpRequest()
+        req.open("POST", '/historyobj')
+        req.send(JSON.stringify(fileData))
+        req.onload = function () {
+            versionHistory = JSON.parse(req.response)
+            docEditor.refreshHistory(versionHistory.hist)
+        }
+    }
```

Comment:
There're two identical code segments. I suggest reusing `onRequestHistory` in `onRequestRestore`.